python/runtime/cudaq/domains/plugins/CMakeLists.txt

```diff
@@ -17,6 +17,6 @@ if (SKBUILD)
     if (NOT Python_FOUND)
       message(FATAL_ERROR "find_package(Python) not run?")
     endif()
-    target_link_libraries(cudaq-pyscf PRIVATE Python::Python pybind11::pybind11 cudaq-chemistry cudaq-spin cudaq cudaq-py-utils)
+    target_link_libraries(cudaq-pyscf PRIVATE Python::Module pybind11::pybind11 cudaq-chemistry cudaq-spin cudaq cudaq-py-utils)
```

Comment:
we should probably take this opportunity to mark the cudaq pyscf create_molecule stuff as deprecated in favor of the cuda-qx implementation.

Reply:
Correct - I don't think this change is needed beyond this library, which we need to deprecate anyway.

Reply:
> we should probably take this opportunity to mark the cudaq pyscf create_molecule stuff as deprecated in favor of the cuda-qx implementation.

It is most likely the necessary thing. My solution on CUDA-QX is a hack that breaks other CUDA-Q build scenarios. (As can be seen in the pipelines here.)